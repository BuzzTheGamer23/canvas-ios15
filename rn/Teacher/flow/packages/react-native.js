// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
