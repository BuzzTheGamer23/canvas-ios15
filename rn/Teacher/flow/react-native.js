//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
