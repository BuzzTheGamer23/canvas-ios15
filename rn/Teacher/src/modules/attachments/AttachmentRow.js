// @flow

import React, { Component } from 'react'
import {
  StyleSheet,
  TouchableHighlight,
  Image,
} from 'react-native'
import Row from '../../common/components/rows/Row'
import images from '../../images'
import colors from '../../common/colors'

export type Props = {
  title: string,
  subtitle: ?string,
  progress: number,
  error: ?string,
  testID: string,
  onRemovePressed: () => void,
  onPress: () => void,
}

export default class AttachmentRow extends Component<any, Props, any> {
  render () {
    return (
      <Row
        title={this.props.title}
        subtitle={this.props.subtitle}
        image={images.attachments.complete}
        imageTint={colors.primaryBrandColor}
        accessories={this.removeButton}
        onPress={this.props.onPress}
        testID={this.props.testID}
      />
    )
  }

  removeButton = (
    <TouchableHighlight
      onPress={this.props.onRemovePressed}
      underlayColor='white'
      hitSlop={{ top: 8, bottom: 8, left: 8, right: 8 }}
    >
      <Image source={images.x} style={styles.remove} />
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  removeContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  remove: {
    width: 14,
    height: 14,
    tintColor: 'black',
  },
})
