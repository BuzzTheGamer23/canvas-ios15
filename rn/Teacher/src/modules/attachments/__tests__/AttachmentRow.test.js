/* @flow */

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import AttachmentRow from '../AttachmentRow'

jest
  .mock('Button', () => 'Button')
  .mock('TouchableHighlight', () => 'TouchableHighlight')
  .mock('TouchableOpacity', () => 'TouchableOpacity')

describe('AttachmentRow', () => {
  it('renders', () => {
    expect(
      renderer.create(
        <AttachmentRow
          title='Attachment 1'
          subtitle='Uploading...'
          onPress={jest.fn()}
          testID='attachment-row.0'
          onRemovePressed={jest.fn()}
        />
      )
    ).toMatchSnapshot()
  })
})
