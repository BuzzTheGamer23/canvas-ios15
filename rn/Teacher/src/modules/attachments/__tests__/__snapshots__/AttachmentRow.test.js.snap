// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentRow renders 1`] = `
<TouchableHighlight
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  onPress={[Function]}
  style={
    Array [
      Object {
        "height": 54,
      },
      undefined,
      undefined,
    ]
  }
  testID="attachment-row.0"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 12.8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12.8,
        },
        Object {
          "backgroundColor": "white",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "marginRight": 16,
            "resizeMode": "contain",
          },
          Object {
            "height": 20,
            "tintColor": "#374A59",
            "width": 20,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Array [
              Object {
                "fontWeight": "600",
              },
              undefined,
            ],
            Object {
              "fontFamily": ".SFUIDisplay-semibold",
            },
          ]
        }
      >
        Attachment 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "color": "#8B969E",
              "fontSize": 14,
              "marginTop": 2,
            },
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
        testID="attachment-row.0-subtitle-lbl"
      >
        Uploading...
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <TouchableHighlight
          hitSlop={
            Object {
              "bottom": 8,
              "left": 8,
              "right": 8,
              "top": 8,
            }
          }
          onPress={[Function]}
          underlayColor="white"
        >
          <Image
            source={1}
            style={
              Object {
                "height": 14,
                "tintColor": "black",
                "width": 14,
              }
            }
          />
        </TouchableHighlight>
      </View>
    </View>
  </View>
</TouchableHighlight>
`;
